search-sentiments-map/routes/search.js

```diff
@@ -0,0 +1,265 @@
+
+/** 
+ * Renders a JSON array of the top search results for all countries with API data
+ * obtained every 12 hours for the specified topic.
+ */
+router.get('/:topic', (req, res) => {
+  let topic = req.params.topic;
+  retrieveSearchResultFromDatastore(topic).then(customSearchTopicJsonArray => {
+    // TODO(ntarn): Remove console.log statements when finished debugging.
+    console.log('ntarn debug: In the router.get method'); 
+    console.log(customSearchTopicJsonArray);
+    res.setHeader('Content-Type', 'application/json');
+    res.send(customSearchTopicJsonArray);
+  });
+});
+
+
+/** 
+ * Returns a JSON-formatted array of search results for countries retrieved
+ * from the Datastore.
+ * @param {string} topic Search topic to get data for.
+ */
+async function retrieveSearchResultFromDatastore(topic) {
+  // Request latest entity with a topic matching the given topic.
+  const query = datastore.createQuery('CustomSearchTopic').order('timestamp', {
+    descending: true,
+  }).filter('topic', topic).limit(1);
+
+  try {
+    const [customSearchTopic] = await datastore.runQuery(query);
+    let customSearchTopicJsonArray = {
+      topic: customSearchTopic[0].topic,
+      countries: customSearchTopic[0].countries,
+      timestamp: customSearchTopic[0].timestamp,
+    };
+    // TODO(ntarn): Remove console.log statements when finished debugging.
+    console.log(`ntarn debug retrieved search result for topic: ${topic}` +  `example: ${customSearchTopic[0].countries[0].results[0]}`);
+    console.log(customSearchTopicJsonArray);
+    return customSearchTopicJsonArray;
+  } catch (err) {
+    console.error('ERROR:', err);
+  }
+}
+
+/** 
+ * Updates daily search results (accumulated by day) in the Datastore.
+ */
+function updateSearchResults() {
+  retrieveGlobalTrends().then(async trends => {
+    // When testing ,use i < 1 to test for only one trend, and comment out
+    // `await new Promise` line to avoid 1 minute pauses.
+    for (let i = 0; i < trends.length; i++) {
+      await updateSearchResultsForTopic(trends[i].trendTopic);
+      await new Promise(resolve => setTimeout(resolve, 60000));
+    }
+  });
+}
+
+/** 
+ * Obtains the most recent global trends by querying the Datastore.
+ * @return {!Array<JSON>} A JSON array of global trends and their originating countries.
+ */
+async function retrieveGlobalTrends() {
+  const query = datastore.createQuery('TrendsEntry').order('timestamp', {
+    descending: true,
+  });
+  const [trendsEntry] = await datastore.runQuery(query);
+  return trendsEntry[0].globalTrends;
+}
+
+/** 
+ * Updates search result data for all countries for a given topic.
+ *   - Deletes stale data from Datastore.
+ *   - Retrieves and formats search result data for each country.
+ *   - Saves the search results for all countries to the datastore.
+ * @param {string} query Search query.
+ */
+async function updateSearchResultsForTopic(query) {
+  let countriesData = [];
+  await deleteAncientResults();
```

Comment:
Actually can we move this line further up into `updateSearchResults` because we don't want to do it for each trend either